updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -89,25 +75,34 @@ def ungrouped_dependencies
 
     private
 
+    sig do
+      params(job: Dependabot::Job, base_commit_sha: String, dependency_files: T::Array[Dependabot::DependencyFile]).void
+    end
     def initialize(job:, base_commit_sha:, dependency_files:)
       @job = job
       @base_commit_sha = base_commit_sha
       @dependency_files = dependency_files
-      @handled_dependencies = Set.new
+      @handled_dependencies = T.let(Set.new, T::Set[String])
 
-      @dependencies = parse_files!
+      @dependencies = T.let(parse_files!, T::Array[Dependabot::Dependency])
```

Comment:
Kind of interesting that Sorbet doesn't infer the types for these methods that are defined in the same file.